src/treemodel/document.js

```diff
@@ -128,38 +148,48 @@ export default class Document {
 	/**
 	 * Creates a new top-level root.
 	 *
+	 * @param {String|Symbol} id Unique root id.
 	 * @param {String|Symbol} name Unique root name.
 	 * @returns {core.treeModel.RootElement} Created root.
 	 */
-	createRoot( name ) {
-		if ( this.roots.has( name ) ) {
+	createRoot( id, name ) {
+		if ( this._roots.has( id ) ) {
 			/**
-			 * Root with specified name already exists.
+			 * Root with specified id already exists.
 			 *
-			 * @error document-createRoot-name-exists
+			 * @error document-createRoot-id-exists
 			 * @param {core.treeModel.Document} doc
-			 * @param {String} name
+			 * @param {String} id
 			 */
 			throw new CKEditorError(
-				'document-createRoot-name-exists: Root with specified name already exists.',
-				{ name: name }
+				'document-createRoot-id-exists: Root with specified id already exists.',
+				{ id: id }
 			);
 		}
 
-		const root = new RootElement( this );
-		this.roots.set( name, root );
+		const root = new RootElement( this, name || id );
+		this._roots.set( id, root );
 
 		return root;
 	}
 
 	/**
-	 * Enqueue a callback with document changes. Any changes to be done on document (mostly using {@link core.treeModel.Document#batch} should
-	 * be placed in the queued callback. If no other plugin is changing document at the moment, the callback will be
+	 * Removes all events listeners set by document instance.
+	 */
+	detach() {
+		this.selection.detach();
+		this.stopListening();
+	}
+
+	/**
+	 * Enqueue a callback with document changes. Any changes to be done on document (mostly using {@link core.treeModel.Document#batch}
```

Comment:
It's details, but it should be "Enqueues". And the fact that it's a callback is a detail which isn't needed in the very first sentence of this method documentation.